### PR TITLE
feat: Standardise table links and tidy event definitions

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

It kept frustrating me that there are subtle differences between a bunch of standard tables. In particular how fiddly it can be to simply click something

## Changes

* Introduce a helper component for `LemonTableLink` s
* Makes the entire space clickable but with the same styling as before
* Removes a lot of cruft and old css
* Accidentally refactored some of the Event/PropertyDefinition areas as it was a mess

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: cypress/e2e/dashboard-deletion.ts
Comment (daibhin) [inline cypress/e2e/dashboard-deletion.ts:1]: I couldn't really verify event editing locally but trusting the two related files were mainly just moving code around.

Comment (daibhin) [inline cypress/e2e/dashboard-deletion.ts:1]: One thing that came to mind is how we might ensure people use the `LemonTableLink` component where it makes sense? Feels like the kind of thing people might not know exists.

Perhaps a Story example would be a good start

**Review by daibhin — APPROVED** [review]
Fantastic stuff - only Ben could have me read 53 files and only have 2 comments. Being able to click anywhere in the cell feels so much better.

File: frontend/__snapshots__/scenes-app-feature-flags--feature-flags-list--light.png
[the remark this answers is not in the document]
Comment (benjackwhite) [reply frontend/__snapshots__/scenes-app-feature-flags--feature-flags-list--light.png:1]: I think its just the diff for one reason or another was never big enough for a snapshot update here until now

File: cypress/e2e/dashboard-deletion.ts
Comment (benjackwhite) [reply cypress/e2e/dashboard-deletion.ts:1]: I think most people discover by copy-paste but I added a story too 👍

**Review by Twixes — COMMENTED** [review]
`LemonTableLink` is :chefs-kiss:!

**Pull request merged**